library/network/test/network_interfaces_helpers_test.rb

```diff
@@ -71,5 +71,47 @@ module Yast
         expect(NetworkInterfaces.filter_interfacetype("INTERFACETYPE" => "dummy")).to include "INTERFACETYPE"
       end
     end
+
+    describe "#get_devices" do
+      let(:data_dir) { File.join(File.dirname(__FILE__), "data") }
+      # MOCKED IN test/data/etc/sysconfig/ifcfg*
+      let(:devices) do
+        ["arc5", "bond0", "br1", "em1", "eth0", "eth1", "ppp0", "tr~", "vlan3"]
+      end
+
+      around do |example|
+        change_scr_root(data_dir, &example)
+      end
+
+      before do
+        subject.main
+        allow(subject).to receive(:Read).and_return(true)
+        allow(Yast::SCR).to receive(:Dir).with(Yast::Path.new(".network.section")).and_return(devices)
+      end
+
+      it "returns an array of configured interfaces filtered by regexp" do
+        expect(subject.get_devices("1")).not_to include "em1", "eth1", "br1"
+      end
+
+      it "filters with <[~]> by default" do
+        expect(subject.get_devices).not_to include "tr~"
+      end
+
+      it "returns an empty array with <.> argument" do
+        expect(subject.get_devices(".")).to eql []
+      end
+
+      it "returns all devices filtering with <''>" do
+        expect(subject.get_devices("")).to eql devices
+      end
+
+      it "does not crash with exception" do
```

Comment:
quite strong statement and you are never sure that it do not return expection. E.g. low memory or signal can get almost in every line of code.